internal/internal_workflow_client.go

```diff
@@ -1658,9 +1654,119 @@ func (w *workflowClientInterceptor) ExecuteWorkflow(
 		dataConverter:    w.client.dataConverter,
 		failureConverter: w.client.failureConverter,
 		registry:         w.client.registry,
+		startRequest:     startRequest,
 	}, nil
 }
 
+func (w *workflowClientInterceptor) executeWorkflowWithOperation(
+	ctx context.Context,
+	startRequest *workflowservice.StartWorkflowExecutionRequest,
+	operation StartWorkflowOperation,
+) (*workflowservice.StartWorkflowExecutionResponse, error) {
+	startOp := &workflowservice.ExecuteMultiOperationRequest_Operation{
+		Operation: &workflowservice.ExecuteMultiOperationRequest_Operation_StartWorkflow{
+			StartWorkflow: startRequest,
+		},
+	}
+
+	var withStartOp *workflowservice.ExecuteMultiOperationRequest_Operation
+	switch t := operation.(type) {
+	case *UpdateWorkflowOperation:
+		if t.WorkflowUpdateHandle != nil {
+			return nil, fmt.Errorf("%w: was already executed", errInvalidWorkflowOperation)
+		}
+
+		updateReq, err := w.createUpdateWorkflowRequest(ctx, t.input)
+		if err != nil {
+			return nil, fmt.Errorf("%w: %w", errInvalidWorkflowOperation, err)
+		}
+		if updateReq.WorkflowExecution.WorkflowId == "" {
+			updateReq.WorkflowExecution.WorkflowId = startRequest.WorkflowId
+		}
+
+		withStartOp = &workflowservice.ExecuteMultiOperationRequest_Operation{
+			Operation: &workflowservice.ExecuteMultiOperationRequest_Operation_UpdateWorkflow{
+				UpdateWorkflow: updateReq,
+			},
+		}
+	default:
+		return nil, fmt.Errorf("unsupported operation request type: %T", t)
+	}
+
+	multiRequest := workflowservice.ExecuteMultiOperationRequest{
+		Namespace: w.client.namespace,
+		Operations: []*workflowservice.ExecuteMultiOperationRequest_Operation{
+			startOp,
+			withStartOp,
+		},
+	}
+	multiResp, err := w.client.workflowService.ExecuteMultiOperation(ctx, &multiRequest)
+
+	// check error and response
+	var multiErr *serviceerror.MultiOperationExecution
+	if err == nil {
+		if len(multiResp.Responses) != len(multiRequest.Operations) {
+			panic(fmt.Sprintf("MultiOperation response from server is incomplete: %v instead of %v",
+				len(multiResp.Responses), len(multiRequest.Operations)))
+		}
+	} else if errors.As(err, &multiErr) {
+		if len(multiErr.OperationErrors()) != len(multiRequest.Operations) {
+			panic(fmt.Sprintf("MultiOperation error response from server is incomplete: %v instead of %v",
+				len(multiErr.OperationErrors()), len(multiRequest.Operations)))
+		}
+	} else {
+		return nil, err
+	}
+
+	// extract errors or responses per operation
+	var startErr error
+	var abortedErr *serviceerror.MultiOperationAborted
+	var startResp *workflowservice.StartWorkflowExecutionResponse
+	for i, opReq := range multiRequest.Operations {
+		switch opReq.Operation.(type) {
```

Comment:
You add a default in your other type switches (pedantic though, I don't mind if you don't here)

Reply:
Right; that's because this switch operates on the previously assembled operations (not the responses). So it would fall into the default if there's a bug in the function earlier. But I can still see it being useful.